include/rnp/rnp2.h

```diff
@@ -0,0 +1,241 @@
+/*-
+ * Copyright (c) 2017 Ribose Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/*
+* Currently we use some definitions/enums from other others, including
+* - rnp_result
+* - pgp_hash_alg_t
+* - pgp_pubkey_alg_t
+*
+* It might be good to consolidate these definitions into one spot, or
+* else replicate them here in new enums and make this header freestanding.
+*/
+#include <rnp_def.h>
+#include <repgp_def.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+/*
+* Opaque structures for key types
+*/
+typedef struct rnp_publickey_st * rnp_public_key;
+typedef struct rnp_privatekey_st *rnp_private_key;
+
+// typedef struct rnp_keyring_st* rnp_keyring;
+
+/* Operations on public keys */
+
+/**
+* Load a public key
+* @param key a pointer, on success will be assigned to the new object
+* @param buf the serialized key (either binary or ASCII formats accepted)
+* @param buf_len length of buf in bytes
+*/
+rnp_result rnp_public_key_load(rnp_public_key *key, const uint8_t buf[], size_t buf_len);
```

Comment:
Similar to the one for loading a private key, the format may need to be a parameter here, right? I mean it could be:
* KBX
* PGP
* PGP (armored) (could auto-detect this though)

Reply:
We could also add a separate `detect` function here, I don't think the FFI library is expected to know all that information.